src/avt/Filters/avtContourFilter.C

```diff
@@ -537,6 +548,10 @@ avtContourFilter::ExecuteDataTree(avtDataRepresentation *in_dr)
                 useVTKm = false;
             }
         }
+
+        //
+        // If we have a 2d structured grid, don't use VTKm.
+        //
```

Comment:
Not part of your change, but I looked at the code for determining if the dataset is 2D, and it only looks at the z-dimension, what would happen if it z-dim > 1 but x or y isn't?  Should the logic be looking at the avtDataAttributes Spatial dimension instead?

Reply:
The conversion works on a vtkDataSet so I don't have access to avtDataAttributes here. I believe my changes will now detect 1d, 2d and 3d grids.

Reply:
The check for 2d rectilinear grid is:
``` 
         int dims[3];
            rgrid->GetDimensions(dims);

            if (dims[2] == 1)
            {
                useVTKm = false;
            }
```
I guess I am confused by what is truly meant by '2D', if dims were {20,1, 20}, isn't that 2D, just in a different plane than if dims were {20, 20, 1}?  VTKM handles the latter but not the former?

Reply:
Sorry, I thought I pushed the changes but didn't. Now it should really be fixed.

Reply:
I thought about this some more and don't think my changes are correct. I think my changes for the structured case are probably correct since I believe the indexing works out if nx=1 or ny=1. I don't think the same change for the rectilinear case is correct though. Was your concern only about structured grids?

Reply:
I was concerned about both cases, and whether or not passing data through VTKm would cause problems when dims[2] > 1, but either dims[0] or dims[1] was 1.  Does the testing still work with your most recent changes?

Reply:
The code will now skip VTKm if any of the dimensions is 1 for the rectlinear and structured case. I checked and VTKm doesn't even handle the 2D topology case if dims[2] == 1. It just gives a blank image.